### PR TITLE
feat: type hint and docstring length

### PR DESCRIPTION
**Pull request by clatapie**
This PR:
- [x] add type hint to each generated function
- [x] fixes most of the issues related to the issues of the docstrings

Some docstrings are still too long. It is most of the time due the fact the line is in a list.

### CODE REVIEW

File: src/pyconverter/xml2py/ast_tree.py [hunk 1]
```diff
@@ -2365,19 +2848,60 @@ def __repr__(self):
...
+    #                     rst_item =  item.to_rst(links=self._links, base_url=self._base_url)
+    #                 elif item.tag in item_needing_fcache:
+    #                     rst_item = item.to_rst(fcache=self._fcache)
+    #                 else:
+    #                     rst_item = item.to_rst()
+    #             else:
+    #                 rst_item = str(item)
+    #             lines.append(rst_item)
+    #         return lines
+    #     return []
```
Comment (RobPasMue) [inline src/pyconverter/xml2py/ast_tree.py:2896]: Remove dead code

File: src/pyconverter/xml2py/ast_tree.py [hunk 2]
```diff
@@ -1920,6 +2117,293 @@ class ProductName(Element):
...
+
+        Returns
+        -------
+        bool
+            True if the argument is an elipsis, False otherwise.
+        """
+        return is_elipsis(str(self._name))
+
+    @property
+    def multiple_args(self):
```
Comment (RobPasMue) [inline src/pyconverter/xml2py/ast_tree.py:2205]: This function is a bit convoluted... it's hard to follow the logic. As long as it works I'm fine with it given the circumstances

**Review by RobPasMue — APPROVED** [review]
Some complex code... I have the feeling it could be simplified but I think it's best we move forward honestly

File: src/pyconverter/xml2py/ast_tree.py [hunk 3]
```diff
@@ -50,6 +51,35 @@
...
+    "second",
+    "third",
+    "fourth",
+    "fifth",
+    "sixth",
+    "seventh",
+    "eighth",
+    "ninth",
+]
+
```
[the remark this answers is not in the document]
Comment (clatapie) [reply src/pyconverter/xml2py/ast_tree.py]: Thanks for raising this. I made some searches and found the [num2words ](https://github.com/savoirfairelinux/num2words)package which seems to fit. I implemented the changes in [9d313e5](https://github.com/ansys/pyconverter-xml2py/pull/340/commits/9d313e5259f225c5db595ec544734f1d38a7876a)

**Review by RobPasMue — APPROVED** [review]

File: pyproject.toml [hunk 4]
```diff
@@ -29,7 +29,9 @@ dependencies = [
     "click>=7.0,<9.0.0",
     "importlib-metadata>=4.0",
     "pygithub>=1.59.1",
+    "inflect>=7.0.0",
     "lxml>=4.9.3",
+    "num2words>=0.5.0",
```
Comment (RobPasMue) [inline pyproject.toml:34]: Hmm I htought we were not using it

**Review by RobPasMue — CHANGES_REQUESTED** [review]

File: pyproject.toml (hunk 4 above)
Comment (clatapie) [reply pyproject.toml:34]: Good catch, I forgot to remove this one
Thanks!

**Review by RobPasMue — APPROVED** [review]
